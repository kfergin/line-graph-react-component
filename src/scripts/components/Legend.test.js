import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Legend from './Legend';

describe('<Legend/>', () => {
	const props = {
		names: [],
		colors: ['#00f']
	};
	
	it('should render its empty snapshot', () => {
		const tree = renderer.create(
			<Legend {...props}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render its snapshot with legend keys', () => {
		const names = ['first', 'second', 'third'];
		const tree = renderer.create(
			<Legend {...{...props, names}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	// is this the right test for the componentWillReceiveProps part?
	it('should set a new timestamp in state when legend keys change', () => {
		const names = ['first', 'second', 'third'];
		const component = mount(
			<Legend {...{...props, names}}/>
		);
		const instance = component.instance();
		const spySetState = jest.spyOn(instance, 'setState');
		component.setProps({names});
		expect(spySetState).not.toHaveBeenCalled();
		component.setProps({names: [
			'fourth', 'fifth', 'sixth'
		]});
		expect(spySetState).toHaveBeenCalled();
	});
});