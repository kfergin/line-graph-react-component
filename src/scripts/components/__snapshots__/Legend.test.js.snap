// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Legend/> should render its empty snapshot 1`] = `
<div
  className="legend"
>
  <ul
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</div>
`;

exports[`<Legend/> should render its snapshot with legend keys 1`] = `
<div
  className="legend"
>
  <ul
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <li
      style={
        Object {
          "color": "#00f",
        }
      }
    >
      first
    </li>
    <li
      style={
        Object {
          "color": "#00f",
        }
      }
    >
      second
    </li>
    <li
      style={
        Object {
          "color": "#00f",
        }
      }
    >
      third
    </li>
  </ul>
</div>
`;
